flow/JsonString.h

```diff
@@ -0,0 +1,142 @@
+
+#ifndef JSONSTRING_H
+#define JSONSTRING_H
+
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+class JsonString;
+class JsonStringArray;
+class JsonStringSetter;
+
+class JsonString {
+	public:
+		JsonString();
+		JsonString( const JsonString& jsonString);
+		JsonString( const JsonStringArray& jsonArray);
+		JsonString( const char* value );
+		JsonString( const std::string& value );
+
+		JsonString( const std::string& name, const char* value );
```

Comment:
These constructor overloads can probably all be handled by a single templated constructor.